src/client-agent/start_agent.c

```diff
@@ -357,6 +363,8 @@ static bool agent_handshake_to_server(int server_id, bool is_startup) {
                         }
 
                         return true;
+                    } else if (strcmp(tmp_msg, HC_INVALID_VERSION) == 0) {
```

Comment:
I think it would be better to make this:
```c
} else if (strncmp(tmp_msg, "err ", 4) == 0) { // Use a define for "err " and strlen for size
    // Parse JSON using cJSON and check message
}
```

Reply:
Done!